datastore/leveldb/leveldbds.nim

```diff
@@ -104,10 +116,18 @@ method query*(
     except Exception as e:
       return failure("Unknown exception in LevelDbDatastore.query -> next: " & $e.msg)
 
-  iter.next = next
-  iter.dispose = proc(): Future[?!void] {.async.} =
+  proc dispose(): Future[?!void] {.async.} =
+    try:
+      dbIter.dispose()
+      return success()
+    except LevelDbException as e:
+      return failure("LevelDbDatastore.query -> dispose exception: " & $e.msg)
+    except Exception as e:
```

Comment:
We shouldn't catch `Exception` and if we do it should be re-raised as a `Defect`.

Reply:
Updated nim-leveldb to correctly show what can be raised by the iterator callbacks.
Adjusted leveldbds.nim to match.